subprojects/osgi-test/osgi-test.gradle

```diff
@@ -30,10 +30,27 @@ dependencies {
 }
 
 test {
+    jvmArgumentProviders.add(
+        new RuntimeBundlesProvider(files: configurations.testRuntimeBundles.asFileTree)
+    )
     dependsOn configurations.testRuntimeBundles
-    inputs.files(sourceSets.testBundle.allSource)
-    inputs.files(sourceSets.otherBundle.allSource)
-    systemProperty 'testRuntimeBundles', configurations.testRuntimeBundles.asPath
+    inputs.files(sourceSets.testBundle.allSource).withPathSensitivity(PathSensitivity.RELATIVE)
```

Comment:
```suggestion
    inputs.files(sourceSets.testBundle.allSource)
        .withPathSensitivity(PathSensitivity.RELATIVE)
        .withPropertyName('testBundleSources')
```

Reply:
The property name will be helpful in distinguishing changes to different properties. Otherwise these appear as `$1`.